### DISCUSSION TITLE
这边想关闭一些怀疑无用的代码，想问一下是否有影响

### DISCUSSION THREAD

**Discussion by zcl1115** [harmonycloud/kindling#607] [agent collector] [answered]
这边的场景只是监测服务或者pod的黄金指标，目前是请求的成功数、失败数、出入流量以及延迟，然后不涉及trace的能力，想要做如下改动看看能不能优化内存的占用:
1. network只监听tcp事件
2. 减少SNAPLEN的大小，缩短到200
3. 减少cgoReceiver的channel大小，由300000缩短到100000
4. 剔除network_analyzer的requestMonitor跟dnsRequestMonitor
特别是最后一个操作是否会有影响，这边并不会用到trace能力

**Comment by dxsup** [comment, marked as answer]
如果完全不需要监控应用层协议的调用次数、响应时间、状态码，那是可以关闭network_analyzer的。但如果需要这些，requestMonitor和dnsRequestMonitor不能剔除，也不能只监听tcp事件。
第2、3是可行的。
